LEGO1/mxticklemanager.cpp

```diff
@@ -0,0 +1,113 @@
+#include "mxomni.h"
+#include "mxticklemanager.h"
+#include "mxtimer.h"
+#include "mxtypes.h"
+
+#include "decomp.h"
+
+#define TICKLE_MANAGER_FLAG_DESTROY 0x1
+
+DECOMP_SIZE_ASSERT(MxTickleClient, 0x10);
+DECOMP_SIZE_ASSERT(MxTickleManager, 0x14);
+
+// OFFSET: LEGO1 0x100bdd10
+MxTickleClient::MxTickleClient(MxCore *p_client, MxTime p_interval)
+{
+  m_flags = 0;
+  m_client = p_client;
+  m_interval = p_interval;
+  m_lastUpdateTime = -m_interval;
+}
+
+// OFFSET: LEGO1 0x100bdd30
+MxTickleManager::~MxTickleManager()
+{
+  while (m_clients.size() != 0) {
+    MxTickleClient *client = m_clients.front();
```

Comment:
Just commenting, not starting a review yet - interesting, I didn't know these things existed in C++.